app/mailers/requests_confirmation_mailer.rb

```diff
@@ -11,20 +11,11 @@ def confirmation_email(request)
   private
 
   def fetch_items(request)
-    combined = combined_items(request)
-    item_ids = combined&.map { |item| item['item_id'] }
-    db_items = Item.where(id: item_ids).select(:id, :name)
-    combined.each { |i| i['name'] = db_items.find { |db_item| i["item_id"] == db_item.id }.name }
-    combined.sort_by { |i| i['name'] }
-  end
+    items_sorted = request.request_items.sort_by { |k| k['item_id'] }
+    item_ids = items_sorted&.map { |item| item['item_id'] }
+    items_names = Item.where(id: item_ids).order(:id).pluck(:name)
+    names_hash = items_names.map { |name| { 'name' => name } }
 
-  def combined_items(request)
-    return [] if request.request_items.size == 0
-    # convert items into a hash of (id => list of items with that ID)
-    grouped = request.request_items.group_by { |i| i['item_id'] }
-    # convert hash into an array of items with combined quantities
-    grouped.map do |id, items|
-      { 'item_id' => id, 'quantity' => items.map { |i| i['quantity'] }.sum }
-    end
+    items_sorted.zip(names_hash).map { |items, names| items.merge(names) }
```

Comment:
This is a pretty obscure chain... can you add a comment to the method indicating what the inputs and outputs are supposed to look like?

Reply:
I added a documentation on the method